src/js/stores/MesosStateStore.js

```diff
@@ -179,10 +187,21 @@ class MesosStateStore extends GetSetBaseStore {
     let services = this.get('lastMesosState').frameworks || [];
     let memberTasks = {};
 
-    services.forEach(function (service) {
+    const schedulerTaskIds = services.reduce((memo, service) => {
+      const serviceName = service.name.split('.').reverse().join('/');
+      const schedulerTask = this.getSchedulerTaskFromServiceName(serviceName);
+
+      if (schedulerTask) {
+        memo.push(schedulerTask.id);
+      }
+
+      return memo;
+    }, []);
+
+    services.forEach((service) => {
       service.tasks.forEach(function (task) {
         if (task.slave_id === nodeID) {
-          memberTasks[task.id] = task;
+          memberTasks[task.id] = isSchedulerTask(task, schedulerTaskIds);
```

Comment:
Ok, now I get it. Thanks for the explanation! Please add a comment explaining why things need to happen in that order and consider changing the data structure as using an array for the IDs is still suboptimal. 
Once we've merged this PR I'd like to have a close look at our `getSchedulerTaskFromServiceName` method, even though that we hopefully burn this store soon, as it is also iterating over the complete list of frameworks and the whole constructs feels a lot like a hack on top of a hack, on top of a hack...